### PR TITLE
feat: add kafka-connect chart

### PR DESCRIPTION
**Pull request by openoms**
Adding a kafka-connect chart, smoketest for dev and testflight

### CODE REVIEW

**Review by bodymindarts — APPROVED** [review]

File: charts/kafka-connect/templates/kafka-connect.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnect
+metadata:
+  name: kafka
+  namespace: {{ .Release.Namespace }}
```
Comment (krtk6160) [inline charts/kafka-connect/templates/kafka-connect.yaml:5]: Does helm not manage this automatically?

File: charts/kafka-connect/templates/kafka-connect.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  config:
+    group.id: connect-cluster
+    offset.storage.topic: connect-cluster-offsets
+    config.storage.topic: connect-cluster-configs
+    status.storage.topic: connect-cluster-status
+    config.storage.replication.factor: -1
+    offset.storage.replication.factor: -1
+    status.storage.replication.factor: -1
+  resources:
+    {{ toYaml (.Values.resources) | nindent 6 }}
```
Comment (krtk6160) [inline charts/kafka-connect/templates/kafka-connect.yaml:22]:
```suggestion
    {{ toYaml .Values.resources | nindent 6 }}
```

File: charts/kafka-connect/values.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+## The image is built with the Dockerfile in /images/kafka-connect
+image:
+  repository: us.gcr.io/galoy-org/kafka-connect
+  digest: "sha256:536348d31fdc138012a93b5ad54761c45075952e7c8d31508e629215e000de18"
+
+## The pods from this namespace are allowed to access the Kafka Connect API
+smoketestNameSpace: "galoy-dev-smoketest"
```
Comment (krtk6160) [inline charts/kafka-connect/values.yaml:7]:
```suggestion
smoketestNamespace: "galoy-dev-smoketest"
```

**Review by krtk6160 — COMMENTED** [review]
just a few nits

File: charts/kafka-connect/templates/kafka-connect.yaml (hunk 1 above)
Comment (openoms) [reply charts/kafka-connect/templates/kafka-connect.yaml:5]: will test again without

File: charts/kafka-connect/values.yaml (hunk 3 above)
Comment (openoms) [reply charts/kafka-connect/values.yaml]: while working on the testflight found that disabling the NetworkPolicy in Kafka altogether is probably the best approach. It adds unnecessary complication and not being used in our other charts either.

File: charts/kafka-connect/templates/kafka-connect.yaml (hunk 1 above)
Comment (openoms) [reply charts/kafka-connect/templates/kafka-connect.yaml]: it works without indeed

**Review by krtk6160 — APPROVED** [review]

**Pull request merged**